packages/react/navigation-menu/src/NavigationMenu.tsx

```diff
@@ -778,12 +765,13 @@ const NavigationMenuContentImpl = React.forwardRef<
     if (context.isRootMenu && content) {
       const handleClose = () => {
         onItemDismiss();
+        onRootContentClose();
```

Comment:
We need to remove from tab order on dismiss in addition to focus outside as the latter doesn't fire unless you're already inside the content when it's dismissed.